S2Cell test failure

### What happened?

The following spec is failing in CI on the `main` branch.

It seems to have originated in this PR https://github.com/CesiumGS/cesium/pull/13581#pullrequestreview-4664583545
but the reason is not clear why as it did not seem to involve relevant code changes.

```
 Core/S2Cell
    ✗ gets correct center of cell
	Expected (-1691083.4101336685, 6039960.550535122, -1153592.7688487188) to equal epsilon (-1691083.4101336703, 6039960.550535122, -1153592.7688487188), 1e-15.
	    at <Jasmine>
	    at UserContext.<anonymous> (packages/engine/Specs/Core/S2CellSpec.js:294:62 <- Build/Specs/SpecList.js:157538:74)
	    at <Jasmine>

2.006 secs: Scene/Scene flyTo camera after morph to CV
1.74 secs: Core/createWorldBathymetryAsync resolves to CesiumTerrainProvider instance with default parameters
1.508 secs: Scene/Scene flyTo camera after morph to 2D
1.332 secs: Scene/Cesium3DTileset Restrict tileset memory usage with maximumCacheOverflowBytes
1.186 secs: Core/TaskProcessor works with a simple worker
1.02 secs: DataSources/KmlDataSource NetworkLink: onInterval
1.005 secs: Scene/Scene flyTo camera after morph to 3D
0.921 secs: Scene/GroundPrimitive larger scene renders batched instances
0.885 secs: Scene/GaussianSplat3DTileContent Load multiple instances of Gaussian splat tileset and validate transformed attributes
0.84 secs: DataSources/PolygonGeometryUpdater Computes dynamic geometry bounding sphere for fill.

Chrome Headless 150.0.0.0 (Linux 0.0.0): Executed 15614 of 15687 (1 FAILED) (skipped 73) (1 min 2.877 secs / 57.397 secs)
TOTAL: 1 FAILED, 15613 SUCCESS


1) gets correct center of cell
     Core/S2Cell
     Expected (-1691083.4101336685, 6039960.550535122, -1153592.7688487188) to equal epsilon (-1691083.4101336703, 6039960.550535122, -1153592.7688487188), 1e-15.
    at <Jasmine>
    at UserContext.<anonymous> (packages/engine/Specs/Core/S2CellSpec.js:294:62 <- Build/Specs/SpecList.js:157538:74)
    at <Jasmine>


TOTAL: 1 FAILED, 15613 SUCCESS
[14:32:55] 'test' errored after 1.35 min
[14:32:55] Error: 1
    at formatError (/home/runner/work/cesium/cesium/node_modules/gulp-cli/lib/versioned/^5.0.0/format-error.js:21:10)
    at Gulp.<anonymous> (/home/runner/work/cesium/cesium/node_modules/gulp-cli/lib/versioned/^5.0.0/log/events.js:33:54)
    at Gulp.emit (node:events:531:35)
    at Gulp.emit (node:domain:489:12)
    at Object.error (/home/runner/work/cesium/cesium/node_modules/undertaker/lib/helpers/createExtensions.js:61:10)
    at handler (/home/runner/work/cesium/cesium/node_modules/now-and-later/lib/map.js:68:20)
    at f (/home/runner/work/cesium/cesium/node_modules/once/once.js:25:25)
    at f (/home/runner/work/cesium/cesium/node_modules/once/once.js:25:25)
    at tryCatch (/home/runner/work/cesium/cesium/node_modules/async-done/index.js:21:15)
    at done (/home/runner/work/cesium/cesium/node_modules/async-done/index.js:37:12)
Error: Process completed with exit code 1.

```

### Reproduction steps

1. Occurring in CI on `main` branch
2.
3.
...


### Sandcastle example

_No response_

### Environment

Browser:
CesiumJS Version:
Operating System:


### AI acknowledgment

- [ ] I used AI to generate this issue report.
- [ ] (If the above is checked) I have reviewed the AI-generated content before submitting.

--- a/packages/engine/Specs/Core/S2CellSpec.js
+++ b/packages/engine/Specs/Core/S2CellSpec.js
@@ -269,35 +269,35 @@ describe("Core/S2Cell", function () {
   it("gets correct center of cell", function () {
     expect(S2Cell.fromToken("1").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(0.0, 0.0),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
     expect(S2Cell.fromToken("3").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(90.0, 0.0),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
     expect(S2Cell.fromToken("5").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(-180.0, 90.0),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
     expect(S2Cell.fromToken("7").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(-180.0, 0.0),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
     expect(S2Cell.fromToken("9").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(-90.0, 0.0),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
     expect(S2Cell.fromToken("b").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(0.0, -90.0),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
     expect(S2Cell.fromToken("2ef59bd352b93ac3").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(105.64131803774308, -10.490091033598308),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
     expect(S2Cell.fromToken("1234567").getCenter()).toEqualEpsilon(
       Cartesian3.fromDegrees(9.868307318504081, 27.468392925827605),
-      CesiumMath.EPSILON15,
+      CesiumMath.EPSILON10,
     );
   });
 

